lib/rouge/lexers/cypher.rb

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Cypher < RegexLexer
+      tag 'cypher'
+      aliases 'cypher'
+      filenames '*.cypher'
+      mimetypes 'application/x-cypher-query'
+
+      title "Cypher"
+      desc 'The Cypher query language (neo4j.com/docs/cypher-manual)'
+
+      def self.functions
+        @functions ||= Set.new %w(
+          ABS ACOS ALLSHORTESTPATHS ASIN ATAN ATAN2 AVG CEIL COALESCE COLLECT
+          COS COT COUNT DATE DEGREES E ENDNODE EXP EXTRACT FILTER FLOOR
+          HAVERSIN HEAD ID KEYS LABELS LAST LEFT LENGTH LOG LOG10 LOWER LTRIM
+          MAX MIN NODE NODES PERCENTILECONT PERCENTILEDISC PI RADIANS RAND
+          RANGE REDUCE REL RELATIONSHIP RELATIONSHIPS REPLACE REVERSE RIGHT
+          ROUND RTRIM SHORTESTPATH SIGN SIN SIZE SPLIT SQRT STARTNODE STDEV
+          STDEVP STR SUBSTRING SUM TAIL TAN TIMESTAMP TOFLOAT TOINT TOINTEGER
+          TOSTRING TRIM TYPE UPPER
+        )
+      end
+
+      def self.predicates
+        @predicates ||= Set.new %w(
+          ALL AND ANY CONTAINS EXISTS HAS IN NONE NOT OR SINGLE XOR
+        )
+      end
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          AS ASC ASCENDING ASSERT BY CASE COMMIT CONSTRAINT CREATE CSV CYPHER
+          DELETE DESC DESCENDING DETACH DISTINCT DROP ELSE END ENDS EXPLAIN
+          FALSE FIELDTERMINATOR FOREACH FROM HEADERS IN INDEX IS JOIN LIMIT
+          LOAD MATCH MERGE NULL ON OPTIONAL ORDER PERIODIC PROFILE REMOVE
+          RETURN SCAN SET SKIP START STARTS THEN TRUE UNION UNIQUE UNWIND USING
+          WHEN WHERE WITH CALL YIELD
+        )
+      end
+
+      state :root do
+        rule %r/[\s]+/, Text
+        rule %r(//.*?$), Comment::Single
+
+        rule %r([*+\-<>=&|~%^]), Operator
+        rule %r/[{}),;\[\]]/, Str::Symbol
+
+        # literal number
+        rule %r/(\w+)(:)(\s*)([._\d]+)/ do
+          groups Name::Label, Str::Delimiter, Text::Whitespace, Num
+        end
+
+        # function-like
+        # - "name("
+        # - "name  ("
+        # - "name ("
+        rule %r/(\w+)(\s*)(\()/ do |m|
+          name = m[1].upcase
+          if self.class.functions.include? name
+            groups Name::Function, Text::Whitespace, Str::Symbol
+          elsif self.class.keywords.include? name
+            groups Keyword, Text::Whitespace, Str::Symbol
+          else
+            groups Name, Text::Whitespace, Str::Symbol
+          end
+        end
+
+        rule %r/:\w+/, Name::Class
+
+        # number range
+        rule %r/(-?\d+)(\.\.)(-?\d+)/ do
+          groups Num, Operator, Num
+        end
+
+        rule %r/(\d+)+/, Num
```

Comment:
Same question here as above about negative numbers?

Reply:
I would say yes. @ElaineRosenberg could you please confirm?

Reply:
Assuming the answer here is the same as above, this should be:

```suggestion
        rule %r/(-?\d+)+/, Num
```

Reply:
Thanks for the suggestion.
I'm waiting for Elaine's answer before making this change.

Reply:
No worries. I've updated the suggestion because for some reason I had a grouping that would have permitted multiple `-` signs.